chainer/training/extensions/polynomial_shift.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import division
+
+from chainer.training import extension
+
+
+class PolynomialShift(extension.Extension):
+
+    """Trainer extension to polynomially shift an optimizer attribute.
+
+    This extension polynomially decreases the specified attribute of the
+    optimizer. The typical use case is an polynomial decay of the
+    learning rate at each iteration.
+
+    For example, suppose that this extension is invoke at every iteration.
+    Then this extension will set the corresponding attribute to
+    ``init_value * (1 - i / max_iter) ^ rate`` at the ``i``-th iteration, where
+    the ``max_iter`` is the number of iterations to be running.
+
+    This extension is also called before the training loop starts by default.
+
+    Args:
+        attr (str): Name of the attribute to shift.
+        (rate, max_count) (float, int): ``rate`` is the the exponent of
+            polynomial shift. ``max_count`` is the number of this extension to
+            be invoked.
+        init (float): Initial value of the attribute. If it is ``None``, the
+            extension extracts the attribute at the first call and uses it as
+            the initial value.
+        target (float): Target value of the attribute. If the attribute reaches
+            this value, the shift stops.
+        optimizer (~chainer.Optimizer): Target optimizer to adjust the
+            attribute. If it is ``None``, the main optimizer of the updater is
+            used.
+
+    """
+    invoke_before_training = True
+
+    def __init__(self, attr, param, init=None, target=None, optimizer=None):
+        self._attr = attr
+        if param[0] < 0:
+            raise ValueError('PolynomialShift does not support negative rate')
+        self._rate = param[0]
+        self._init = init
+        self._target = target
+        self._optimizer = optimizer
+        self._t = 0
+        self._max_iter = param[1]
+
```

Comment:
How about maintaining/serializing/deserializing `_last_value` as in `LinearShift` and `ExponentialShift`?
It's used to re-initialize the optimizer on deserialization.